src/tojson.jl

```diff
@@ -4,53 +4,117 @@ using JSON
 """
     tojson(output::String, ts::ReportingTestSet)
 
-Takes user output and a ReportingTestSet and converts it to a JSON string as expected by the interface.
+Takes user output and a ReportingTestSet and converts it to a JSON string as
+expected by the interface.
+
+Here's a brief summary of the schema we're following:
+
+{
+    version: 2
+    status: pass | fail | error
+    message?: "a descriptive message emitted only if none of the tests ran (e.g. a syntax error in the file)"
+    tests: [
+        {
+            name: "a human-readable name"
+            status: pass | fail | error
+            message?: "Describes the result of the test if it did not pass"
+            output?: "the stdout for this test"
+            test_code?: "snippet of code that failed that student can try locally"
+                (required for failing or erroring concept exercise tests)
+            task_id?: "a concept exercise thing we don't support yet"
+                (bump version to 3 if we support this)
+        },
+    ]
+}
+
+Note:
+    We currently populate test_code only with the failing or erroring `@test`
+    expression. This means that we don't give all the necessary information if
+    the test relies on definitions outside of that expression, e.g.
+
+    ```julia
+    @testset begin
+        robot = Robot()
+        @test robot.rename!()
+    end
+    ```
+
+    For that to make sense, the student would really need the entirety of the
+    enclosing testset, but we will only give them the string
+    `@test robot.rename!()`
```

Comment:
This bit is important. If we write the tests appropriately then this is fine.

If not, we'll have to do the metaprogramming I mentioned in #26 that I couldn't be bothered to do today.